src/nodeattr/nodeattr.c

```diff
@@ -1000,6 +971,70 @@ _hash_attrval(hash_t hattr, char *node, char *attr, char *val)
     }
 }
 
+static void 
+list_attrs(genders_t gp, char *node)
+{
+    char **attrs, **vals;
+    char *anode = NULL;
+    int numattrs, len, vlen, count, i;
+    hash_t hattrval = NULL;
+    hostlist_t nodelist;
+
+    if ((len = genders_attrlist_create(gp, &attrs)) < 0)
+        _gend_error_exit(gp, "genders_attrlist_create");
+    if ((vlen = genders_vallist_create(gp, &vals)) < 0)
+        _gend_error_exit(gp, "genders_vallist_create");
+    if ((numattrs = genders_getnumattrs(gp)) < 0)
+        _gend_error_exit(gp, "genders_getnumattrs");
+    if (!(hattrval = hash_create((numattrs + 1)*4,

```

Comment:
comment to explain the `(numattrs +1)*4` would be good.  I think it's from elsewhere in nodeattr.

Reply:
yes, repeated from elsewhere.  I copied the comments now too